viper.go

```diff
@@ -313,6 +313,8 @@ func (v *Viper) searchMap(source map[string]interface{}, path []string) interfac
 
 	if next, ok := source[path[0]]; ok {
 		switch next.(type) {
+		case map[interface{}]interface{}:
```

Comment:
Ironically I had taken care of this edge case in the previous (unmerged) implementation. Good catch.